packages/truffle-core/lib/commands/develop.js

```diff
@@ -14,102 +15,83 @@ var command = {
     usage: "truffle develop",
     options: []
   },
-  runConsole: function(config, ganacheOptions, done) {
-    var Console = require("../console");
-    var Environment = require("../environment");
+  async runConsole(config, ganacheOptions, done) {
+    const Console = require("../console");
+    const Environment = require("../environment");
 
-    var commands = require("./index");
-    var excluded = ["console", "init", "develop"];
+    const commands = require("./index");
+    const excluded = ["console", "develop", "unbox", "init"];
 
-    var available_commands = Object.keys(commands).filter(function(name) {
-      return excluded.indexOf(name) === -1;
-    });
+    const available_commands = Object.keys(commands).filter(
+      name => !excluded.includes(name)
+    );
 
-    var console_commands = {};
-    available_commands.forEach(function(name) {
+    const console_commands = {};
+    available_commands.forEach(name => {
       console_commands[name] = commands[name];
     });
 
-    Environment.develop(config, ganacheOptions, function(err) {
-      if (err) return done(err);
-
-      var c = new Console(
-        console_commands,
-        config.with({
-          noAliases: true
-        })
-      );
+    const environmentDevelop = util.promisify(Environment.develop);
+    environmentDevelop(config, ganacheOptions).catch(err => done(err));
 
-      c.start(done);
-      c.on("exit", function() {
-        process.exit();
-      });
+    const c = new Console(console_commands, config.with({ noAliases: true }));
+    c.start(done);
+    c.on("exit", async () => {
+      process.exit();
     });
   },
-  run: function(options, done) {
-    var Config = require("truffle-config");
-    var Develop = require("../develop");
-
-    var config = Config.detect(options);
-    var customConfig = config.networks.develop;
-    let numAddresses = 10;
-    let defaultEtherBalance = 100;
-    let bTime = 0;
-
-    if (customConfig) {
-      numAddresses = customConfig.accounts ? customConfig.accounts : 10;
-      defaultEtherBalance = customConfig.defaultEtherBalance
-        ? customConfig.defaultEtherBalance
-        : 100;
-      bTime = customConfig.blockTime ? customConfig.blockTime : 0;
-    }
+  async run(options, done) {
+    const Config = require("truffle-config");
+    const Develop = require("../develop");
```

Comment:
For the above 2 lines:

Non-standard variable naming with leading capital. See above, feel free to ignore/resolve.